src/cryptonote_core/tx_pool.h

```diff
@@ -62,24 +65,33 @@ namespace cryptonote
   //! pair of <transaction fee, transaction hash> for organization
   typedef std::pair<std::pair<double, std::time_t>, crypto::hash> tx_by_fee_and_receive_time_entry;
 
-  class txCompare
+  class txFeeCompare
   {
   public:
-    bool operator()(const tx_by_fee_and_receive_time_entry& a, const tx_by_fee_and_receive_time_entry& b) const
+    bool operator()(const std::pair<double, std::time_t>& a, const std::pair<double, std::time_t>& b) const
     {
       // sort by greatest first, not least
-      if (a.first.first > b.first.first) return true;
-      if (a.first.first < b.first.first) return false;
+      if (a.first > b.first) return true;
+      if (a.first < b.first) return false;
 
-      if (a.first.second < b.first.second) return true;
-      if (a.first.second > b.first.second) return false;
+      if (a.second < b.second) return true;
+      return false;
+    }
+  }; 
 
-      return memcmp(a.second.data, b.second.data, sizeof(crypto::hash)) < 0;
+  class hashCompare
+  {
+  public:
+    bool operator()(const crypto::hash& a, const crypto::hash& b) const
+    {
+      return memcmp(a.data, b.data, sizeof(crypto::hash)) < 0;
     }
   };
 
   //! container for sorting transactions by fee per unit size
-  typedef std::set<tx_by_fee_and_receive_time_entry, txCompare> sorted_tx_container;
+  typedef boost::bimap<boost::bimaps::multiset_of<std::pair<double, std::time_t>, txFeeCompare>,
+                       boost::bimaps::set_of<crypto::hash, hashCompare>> sorted_tx_container;
```

Comment:
What about using `unordered_set_of` here? This could accelerate the lookup, but would likely require more memory (for the table), and perhaps resizing the table would be slow.

Reply:
What do you think? At this point it comes to the design choices, I am perfectly happy to change it to `unordered_set_of`.

Reply:
Search times are a bit lower with an `unordered_map`, but insertion times may be slower with a large number of keys. I think we have more searches than insertions, but I am not certain. So just leave it for now, unless. you want to run another test.

Reply:
Thanks a lot for detailed explanation.